src/Storages/StorageMaterializeMySQL.cpp

```diff
@@ -92,7 +92,7 @@ Pipe StorageMaterializeMySQL::read(
     {
         Block pipe_header = pipe.getHeader();
         auto syntax = TreeRewriter(context).analyze(expressions, pipe_header.getNamesAndTypesList());
-        ExpressionActionsPtr expression_actions = ExpressionAnalyzer(expressions, syntax, context).getActions(true);
+        ExpressionActionsPtr expression_actions = ExpressionAnalyzer(expressions, syntax, context).getActions(true, false);
```

Comment:
```suggestion
        ExpressionActionsPtr expression_actions = ExpressionAnalyzer(expressions, syntax, context).getActions(true /* add_aliases */, false /* project_result */);
```

Not that it becomes more clear, but at least the next level of unclear...

By the way, what would it mean for a filter transform to "project result"? Our use of "projection" is a misnomer, I think we use it to refer to a final rename of columns performed at some latest stage of query execution. Can FilterTransform do this rename as well?

Reply:
AFAIK `project_result` means "leave only result columns of `expressions` in block, remove other columns (including columns with intermediate results of some subexpressions)". Since we added all required columns into `expressions`, we can expect that it will work fine with `project_result = true`. But `StorageMergeTree::read(require_columns_name, ...)` may return some extra columns for `PREWHERE` (for example, `equals(brand_id, 15837)` from the linked issue) even if nobody asks for such columns. Interpreter does not explicitly add these columns into `require_columns_name`, but expects that `read(...)` will understand somehow that it must returns special columns for `PREWHERE` in addition to required columns... 

Seems like it's not possible to make this code more clear without rewriting everything around `PREWHERE`.